application/src/test/java/org/thingsboard/server/service/state/DefaultDeviceStateServiceTest.java

```diff
@@ -294,6 +294,7 @@ public void decreaseInactivityForActiveDeviceTest() throws Exception {
         Mockito.verify(telemetrySubscriptionService, Mockito.never()).saveAttrAndNotify(Mockito.any(), Mockito.eq(deviceId), Mockito.any(), Mockito.eq("active"), Mockito.any(), Mockito.any());
 
         long newTimeout = 1;
+        Thread.sleep(newTimeout);
```

Comment:
This test fails 100% in the same millisecond. As we saw when we mocked the system time wrapper.

I agree with your suggestion to add another test. But in the parametrized way.
AtomicLong atomicTs = new AtomicLong(); // in the Setup section

Lets mock `willAnswer (() -> atomicTs.addAndGet(msFromParameter)).given(service).currentTimeMs()`
 msFromParameter=0 - you got a frozen time
 msFromParameter=1 - you got controlled time each call will return the next ms (this is default mock)